www/app/hardware/Hardware.html

```diff
@@ -979,6 +979,39 @@ <h2><span data-i18n="Hardware">Hardware</span>:</h2>
 			</tr>
 		</table>
 	</div>
+	<div id="divopenweathermap">
+		<br>
+		<table class="display" id="hardwareparamsopenweathermap" border="0" cellpadding="0" cellspacing="20">
+			<tr>
+				<td align="right" style="width:110px"><label><span data-i18n="API Key"></span>:</label></td>
+				<td><input type="text" id="apikey" style="width: 250px; padding: .2em;" class="text ui-widget-content ui-corner-all" /></td>
+			</tr>
+			<tr>
+				<td align="right" style="width:110px"><label><span data-i18n="Location"></span>:</label></td>
+				<td>
+					<input type="text" id="location" style="width: 250px; padding: .2em;" class="text ui-widget-content ui-corner-all" /><br />
+					<span data-i18n="StationLatLong"></span><br />
+					<span data-i18n="(0 = Application default)"></span>
+				</td>
+			</tr>
+			<tr>
+				<td align="right" style="width:110px"><label for="addcurrent"><span data-i18n="Current Weather">Current Weather</span>:</label></td>
+				<td><input type="checkbox" id="addcurrent" checked="checked" />&nbsp;<label for="addcurrent">&nbsp;</label></td>
+			</tr>
+			<tr>
+				<td align="right" style="width:110px"><label for="adddayforecast"><span data-i18n="Daily forecast">Daily forecast</span>:</label></td>
+				<td><input type="checkbox" id="adddayforecast" />&nbsp;<label for="adddayforecast">&nbsp;Forecast for today and coming 7 days  (if available).</label></td>
+			</tr>
+			<tr>
+				<td align="right" style="width:110px"><label for="addhourforecast"><span data-i18n="Hourly forecast">Hourly forecast</span>:</label></td>
+				<td><input type="checkbox" id="addhourforecast" />&nbsp;<label for="addhourforecast">&nbsp;Forecast for each hour for 48 hours (if available).</label></td>
+			</tr>
+			<tr>
+				<td align="right" style="width:110px"><label for="pollinterval"><span data-i18n="Poll Interval"></span>:</label></td>
```

Comment:
I don't think this should be a user defined option here. It might cause hammering and could result in a ban.
It also does not make any sense to poll faster then once every 5 minutes

Reply:
Good point, Did considered it also and there is some sanity checking the values in the constructor (around line 80) with a minimum of 10 seconds and max of 86400 (a day). OWM has a limit of 60 per second! and 1mln per day :) But agree that some sensible values are needed.

I will set the minimum to 60 seconds and default remains 600 seconds.